fiftyone/plugins/core.py

```diff
@@ -90,16 +90,27 @@ def disable_plugin(plugin_name):
 
 
 def delete_plugin(plugin_name):
-    """Deletes the given plugin from local disk.
+    """Deletes the given plugin and its settings from local disk.
 
     Args:
         plugin_name: the plugin name
     """
-    plugin = _get_plugin(plugin_name)
+    plugin = next(_get_plugin(plugin_name))
     _update_plugin_settings(plugin_name, delete=True)
     etau.delete_dir(plugin.path)
 
 
+def delete_plugin_dir(plugin_dir):
```

Comment:
I would rather not have a `delete_plugin_dir()` method because it may cause unnecessary confusion about what the difference between a plugin and a plugin directory actually is.

If you do have duplicate plugins, it may be because you have the same plugin on your path twice, but it also could be that you're developing a plugin whose name clashes with another plugin, in which case the solution is not to delete either plugin but instead rename one plugin. Or maybe that's not the solution, but point is that dealing with duplicate plugins is a manual workflow.

Reply:
I tried to use the docstring to make it clear what the use case is and why it's different from deleting a plugin.

I agree that dealing with duplicate plugins is manual, so that's why I didn't make a delete duplicate plugin method. Although a user can easily delete a directory in other ways, I think this is just nice to have to fully resolve a duplicate plugin error without using another library or going into the gui filesystem